e2e/src/sozu/worker.rs

```diff
@@ -0,0 +1,255 @@
+            accept_queue_timeout: None,
+        }
+    }
+    pub fn empty_listeners() -> Listeners {
+        Listeners {
+            http: Vec::new(),
+            tls: Vec::new(),
+            tcp: Vec::new(),
+        }
+    }
+    pub fn into_config(config: FileConfig) -> Config {
+        config.into("").expect("could not create Config")
+    }
+    pub fn empty_config() -> (Config, Listeners) {
+        let listeners = Worker::empty_listeners();
+        let config = Worker::empty_file_config();
+        let config = Worker::into_config(config);
+        (config, listeners)
+    }
+
+    pub fn create_server(config: Config, listeners: Listeners) -> Server {
+        let (scm_main, scm_worker) = UnixStream::pair().expect("could not create unix stream pair");
+        let (_, cmd_worker) =
+            Channel::generate(config.command_buffer_size, config.max_command_buffer_size)
+                .expect("could not create a channel");
+
+        let scm_main = ScmSocket::new(scm_main.as_raw_fd());
+        scm_main
+            .send_listeners(&listeners)
+            .expect("could not send listeners");
+
+        Server::try_new_from_config(
+            cmd_worker,
+            ScmSocket::new(scm_worker.as_raw_fd()),
+            config,
+            ConfigState::new(),
+            false,
+        )
+        .expect("could not create sozu worker")
+    }
+
+    pub fn start_new_worker(config: Config, listeners: Listeners) -> Self {
+        let (scm_main, scm_worker) = UnixStream::pair().expect("could not create unix stream pair");
+        let (cmd_main, cmd_worker) =
+            Channel::generate(config.command_buffer_size, config.max_command_buffer_size)
+                .expect("could not create a channel");
+
+        let scm_main = ScmSocket::new(scm_main.as_raw_fd());
+        scm_main
+            .send_listeners(&listeners)
+            .expect("could not send listeners");
+
+        let job = thread::spawn(move || {
+            let mut server = Server::try_new_from_config(
+                cmd_worker,
+                ScmSocket::new(scm_worker.as_raw_fd()),
+                config,
+                ConfigState::new(),
+                false,
+            )
+            .expect("could not create sozu worker");
+            server.run();
+        });
+
+        Self {
+            scm_socket: scm_main,
+            command_channel: cmd_main,
+            command_id: CommandID::new(),
+            job,
+        }
+    }
+
+    pub fn send_proxy_request(&mut self, order: ProxyRequestOrder) {
+        self.command_channel.write_message(&ProxyRequest {
+            id: self.command_id.next(),
+            order,
+        });
+    }
+    pub fn read_proxy_response(&mut self) -> Option<ProxyResponse> {
+        self.command_channel.read_message()
+    }
+    pub fn read_to_last(&mut self) {
+        loop {
+            let response = self.read_proxy_response();
+            println!("test received: {:?}", response);
```

Comment:
maybe implement `std::fmt` for response so that the output is more cosmetic than:

```
test received: Some(ProxyResponse { id: "ID_1", status: Ok, content: None })

```